stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -396,6 +396,85 @@ function _rdiv!(Dc::Diagonal, Db::Diagonal, Da::Diagonal)
 end
 ldiv!(Dc::Diagonal, Da::Diagonal, Db::Diagonal) = Diagonal(ldiv!(Dc.diag, Da, Db.diag))
 
+# optimizations for (Sym)Tridiagonal and Diagonal
+function (\)(D::Diagonal, S::SymTridiagonal)
+    T = typeof(oneunit(eltype(D)) \ oneunit(eltype(S)))
+    # this is essentially copy_oftype, but we need to handle the case when S.ev has the
+    # same length as S.dv
+    dl = copyto!(similar(S.ev, T, length(S.dv)-1), _evview(S))
+    d  = copy_oftype(S.dv, T)
+    du = copy(dl)
+    ldiv!(D, Tridiagonal(dl, d, du))
+end
+function (\)(D::Diagonal, T::Tridiagonal)
+    Td = typeof(oneunit(eltype(D)) \ oneunit(eltype(T)))
+    ldiv!(D, copy_oftype(T, Td))
```

Comment:
We can call `ldiv!(similar(D,Td), D, T)` here to avoid copy. (Applies to `rdiv!`)